src/SSO/SingleSignOn.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+
+namespace SSO;
+
+
+class SingleSignOn
+{
+    protected $company;
+    protected $key;
+    protected $userData;
+
+    const GET_URL_APP = 'https://app.klipfolio.com/users/sso_auth?sso=%s&company=%s&redirect=true';
+    const GET_URL_SB = 'https://sandbox.klipfolio.com/users/sso_auth?sso=%s&company=%s&redirect=true';
+
+    public function __construct($company, $key, $userData) {
+        $this->company = $company;
+        $this->key = $key;
+        $this->userData = $userData;
+    }
+
+    public function generateToken()
+    {
+        $salted = $this->key . $this->company;
+        $hash = hash('sha1', $salted, true);
+        $saltedHash = substr($hash, 0, 16);
```

Comment:
`$saltedHash` is actually the key you're using for the encryption, correct? Perhaps you should name it more appropriately.